packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriod.component.js

```diff
@@ -0,0 +1,127 @@
+import { useMemo } from 'react';
+
+import { format, isValid } from 'date-fns';
+import isEmpty from 'lodash/isEmpty';
+import PropTypes from 'prop-types';
+
+import Badge from '@talend/react-components/lib/Badge';
+
+import { operatorPropTypes, operatorsPropTypes } from '../../facetedSearch.propTypes';
+import { BadgeFaceted } from '../BadgeFaceted';
+import { BadgePeriodForm } from './BadgePeriodForm.component';
+
+const DATE_FORMAT_YYYY_DD_MM = 'yyyy-MM-dd';
+
+function isDateRange(value) {
+	return value.id === 'CUSTOM';
+}
+
+function formatDate(date) {
+	return format(date, DATE_FORMAT_YYYY_DD_MM);
+}
+
+function getRangeLabel(startDateTime, endDateTime, t) {
+	return `${formatDate(startDateTime)} ${t('TO', { defaultValue: 'to' })} ${formatDate(
+		endDateTime,
+	)}`;
+}
+
+const getSelectBadgeLabel = (value, t) => {
+	const labelAll = t('FACETED_SEARCH_VALUE_ALL', { defaultValue: 'All' });
+	if (!isEmpty(value)) {
+		if (isDateRange(value)) {
+			if (isValid(value.startDateTime) && isValid(value.endDateTime)) {
+				return getRangeLabel(value.startDateTime, value.endDateTime, t);
+			}
+			return labelAll;
+		}
+		return value.label;
+	}
+	return labelAll;
+};
```

Comment:
I would like to suggest following thing as it is more readable and maintainable (only two "return", no if cascade)

```suggestion
const getSelectBadgeLabel = (value, t) => {
	const labelAll = t('FACETED_SEARCH_VALUE_ALL', { defaultValue: 'All' });

	if (isEmpty(value) || !isValid(value.startDateTime) || !isValid(value.endDateTime) return labelAll;
	
	return isDateRange(value)
		? getRangeLabel(value.startDateTime, value.endDateTime, t)
		: value.label;
};
```

Reply:
Simply the function to have less `if` conditions.